### PR TITLE
Jetpack connect: Use global notices

### PR DESCRIPTION
**Pull request by sirreal**
Includes #24428 

This is a proof-of-concept which modifies `JetpackConnectNotices` component to dispatch redux notice actions rather than render components.

This moves notices which were previously contained inside the Jetpack Connect dialog out to be Global Notices. Here's a before/after example:

| Before | After |
| --- | --- |
| before | after |

This is relevant as we work on optimistic plans rendering. It will be important to be able to show notices, such as status updates, that are not linked to container components. For example, we'll want to show connection updates, while allowing the user to select a plan.

This is just an initial implementation and I'd like to get design feedback before going any further.

Try playing with this on [calypso.live](https://calypso.live/jetpack/connect?branch=update/jpc-auth-global-notices)

### CODE REVIEW

**Comment by folletto** [comment]
For this kind of actions we have a pattern that has error messages below the field, like:

screen shot 2018-04-24 at 19 12 55

Ref: https://wpcalypso.wordpress.com/devdocs/design/form-fields

Any reason I'm missing for not doing that?

**Comment by sirreal** [comment]
@folletto For that case, the form validation type message is probably a good fit. The notice dispatches an action on dismiss, I'm not sure if that's really needed or how that could work with the form validation type messages, but we could investigate.

This is likely a better example of what we're trying to achieve. Please don't pay extremely close attention to the experience, it's very rough with a forced error. However the idea is there, that we're trying to get plans in front of the user that they can interact with, while the Jetpack connection process continues in the background. For these cases, I believe it makes sense to pull the notices out of the components so they're not tied to a particular layout and can persist across navigation and page content changes. Here's what I mean:

Before (tied to component/layout)

notices-static | 

After (global)

notices

You might imagine messages like "Thanks for picking a plan, we're finishing up…" or "Your is now connected, please pick a plan (or free) to continue." also appearing in this flow. I'd like to hear your thoughts here 🙂

**Comment by folletto** [comment]
I think I get what you mean – is it a matter  of allowing and communicating a background task while another action is being performed, right?

From a technical standpoint I get it, but I feel the choice has to be tied to specific instances... and in the instances above it seems to me that's not a sensible choice:

1. "That's not a valid URL" is form validation and should block the action (button disabled?)
2. The sequence of error attempts asynchronously in the demo above should probably be reviewed more accurately. For instance, I'm not sure we should notify of retries if the retry is automatic, until we actually fail and require user intervention — but I'm not aware of that specific scenario, so don't take this for a proper advice.

tl;dr: it seems this needs a deeper design review on the specific flow.

**Comment by joanrho** [comment]
It does make sense to separate notices that aren't linked to container components from ones that are, but I agree with @folletto that we need to consider the specific flows for each scenario. 

For ex, in our remote install notice work, we've introduced "friendlier" notices with clear next steps for users to take when an error occurs—e.g., credentials are incorrect where form fields are highlighted or the user needs a WP version update (see p7rd6c-1dS#comment-2247), user does not have Admin privileges to install plugins, etc. (see p7rd6c-1dS#comment-2239).

You mention messages like "Thanks for picking a plan, we're finishing up…" or "Your is now connected, please pick a plan (or free) to continue." — for both of these, it'd make more sense to have a message on the page with clear next steps to take, perhaps like the above remote install examples. We'll soon be implementing an updated Plan > My Plan page (checklist) that has a new "Thank you!" splash with illustration and progress (see p6TEKc-1UX#comment-4743), which relates to the first message scenario.

I'm a little unclear on your "After (global)" example—you're starting at the "Approve" step of the connection to wpcom, and then you're quickly taken to the Plans grid. Then there's a "In some cases, Authorization can take a few attempts. Please try again." notice up top, but it's not clear what the user should try again. Do they go back in their browser window? Would it be possible to just keep the user on the authorization/connection view until the authorization is complete?

**Comment by sirreal** [comment]
Thanks both for your thoughts 👍 

I think I agree with pretty much everything that's been said. I'll incorporate this feedback in some other work that's brewing.
